libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.cpp

```diff
@@ -286,28 +286,29 @@ bool AP_ExternalAHRS_VectorNav::check_uart()
     return true;
 }
 
-// Send command to read given register number and wait for response
+// Send command and wait for response
 // Only run from thread! This blocks until a response is received
```

Comment:
```suggestion
// Only run from thread! This blocks and retries until a non-error response is received
```

What if the error is permanent?

Reply:
As discussed above, if the error is permanent then the system will continue to attempt to send the command every 0.5 seconds, receiving a VNERR response and sending that to GCS. In these cases, initialization will not complete and the unit will not arm.